be/src/exec/pipeline/exchange/local_exchange_source_operator.cpp

```diff
@@ -113,12 +115,17 @@ ChunkPtr LocalExchangeSourceOperator::_pull_shuffle_chunk(RuntimeState* state) {
         while (!_partition_chunk_queue.empty() &&
                rows_num + _partition_chunk_queue.front().size <= state->chunk_size()) {
             rows_num += _partition_chunk_queue.front().size;
+            memory_size += _partition_chunk_queue.front().chunk->bytes_usage(_partition_chunk_queue.front().from,
+                                                                             _partition_chunk_queue.front().size);
```

Comment:
This shuffled chunk doesn't uses `[from, size)`, but the indexes `partition_chunk.indexes->data()[from]`, `partition_chunk.indexes->data()[from+1]`, ...`partition_chunk.indexes->data()[from + size -1 ]`.

Reply:
my mistake, fixed by actually computing partition memory usage.